model-server/src/main/kotlin/org/modelix/model/server/handlers/RepositoriesManager.kt

```diff
@@ -234,12 +236,12 @@ class RepositoriesManager(val client: LocalModelClient) : IRepositoriesManager {
             for (branchName in getBranchNames(repository)) {
                 putVersionHash(repository.getBranchReference(branchName), null)
             }
-            store.getGenericStore().put(branchListKey(repository), null)
+            genericStore.put(branchListKey(repository), null)
             val isolated = checkNotNull(isIsolated(repository)) { "Repository not found: $repository" }
             val existingRepositories = getRepositories(isolated)
             val remainingRepositories = existingRepositories - repository
             store.put(repositoriesListKey(isolated), remainingRepositories.joinToString("\n") { it.id })
-
+            genericStore.removeRepositoryObjects(repository)
```

Comment:
How expensive is it to do all of this inside the transaction?

Reply:
I don't know how to quantify that. In the long run repository objects should be stored in a separate cache without transactions, as the data is conceptually immutable (but removable).

Reply:
How long does it take for our big customer? The question is if it's valid to do all that in the transaction and also in the scope of the REST request.

Reply:
Haven't run into issues yet. We can follow up on it, if it becomes a problem.